src/app/profile/receipts/receipts.component.js

```diff
@@ -0,0 +1,99 @@
+import angular from 'angular';
+import template from './receipts.tpl';
+import donationsService from 'common/services/api/donations.service';
+import filterByYear from './receipts.filter';
+import loadingOverlay from 'common/components/loadingOverlay/loadingOverlay.component';
+import sessionEnforcerService from 'common/services/session/sessionEnforcer.service';
+import {Roles} from 'common/services/session/session.service';
+import uniq from 'lodash/uniq';
+import sortBy from 'lodash/sortBy';
+
+class ReceiptsController {
+
+  /* @ngInject */
+  constructor(donationsService,sessionEnforcerService,$location,$window,$log) {
+    this.donationsService = donationsService;
+    this.sessionEnforcerService = sessionEnforcerService;
+    this.$location = $location;
+    this.$window = $window;
+    this.$log = $log;
+    this.loading = false;
+    this.years = [];
+    this.showYear = '';
+    this.maxShow = this.step = 25;
+    this.retrievingError = '';
+  }
+
+  $onInit() {
+    this.enforcerId = this.sessionEnforcerService([Roles.registered], {
+      'sign-in': () => {
+        this.$window.location = '/receipts.html';
+      },
+      cancel: () => {
+        this.$window.location = 'cart.html';
+      }
+    });
+    this.getReceipts();
+  }
+
+  getReceipts(){
+    this.loading = true;
+    let today = new Date();
+    let currentYear = today.getFullYear()*1;
+
+    let data = {
+      'end-date': {
+
+      },
+      'start-date': {
+        'display-value': currentYear - 10 + '-' + ((today.getMonth()*1)+1) + '-' + today.getDate()
+      }
+    };
+    this.donationsService.getReceipts(data)
```

Comment:
How long does fetching the last 10 years take? This may need to be refactored to fetch a year at a time, or as the user selects a year from the UI. Some users may have upwards of a few hundred receipts a year.